include/revng/PipelineC/PipelineC.h

```diff
@@ -0,0 +1,378 @@
+#else
+typedef struct rp_kind rp_kind;
+#endif
+
+#ifdef __cplusplus
+typedef pipeline::Step rp_step;
+#else
+typedef struct rp_step rp_step;
+#endif
+
+#ifdef __cplusplus
+typedef pipeline::ContainerSet::value_type rp_container_checkpoint;
+#else
+typedef struct rp_container_checkpoint rp_container_checkpoint;
+#endif
+
+#ifdef __cplusplus
+typedef const pipeline::ContainerFactorySet::value_type rp_container;
+#else
+typedef struct rp_container rp_container;
+#endif
+
+#ifdef __cplusplus
+typedef const pipeline::Target rp_target;
+#else
+typedef struct rp_target rp_target;
+#endif
+
+#ifdef __cplusplus
+typedef const pipeline::TargetsList rp_container_status;
+#else
+typedef struct rp_container_status rp_container_status;
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Every pointer returned by this library is owned by the library and must not
+ * be deleted, except for pointers returned by functions containing the word
+ * "create". Objects returned by functions named "create" are owned by the
+ * caller and must be destroyed invoking the appropriate function called
+ * rp_TYPE_destroy.
+ */
+
+/**
+ * Must be invoked before any other rp_* function is used,
+ * must be invoked exactly once.
+ * This will take care of initializing all llvm related
+ * stuff needed by the revng pipeline to operate.
+ *
+ * Do not initialize all stuff on your own.
+ */
+bool rp_init(int argc,
+             char *argv[],
+             int libraries_count,
+             const char *libraries_path[]);
+
+/**
+ * pipeline_path, flags, execution_directory cannot be null.
+ * execution_directory can be empty, if it is empty then the content of the
+ * pipeline will not be loaded and saved on disk before and after the execution.
+ * pipeline_flags can be empty, pipeline_flags_count must be the size of the
+ * pipeline_flags array.
+ *
+ * returns the created rp_runner if no error happened, nullptr otherwise.
+ *
+ * this function can be called only once, since it will take take of the
+ * initialization of all dynamically loaded libraries.
+ *
+ * this must be invoked after rp_init and all invocations of
+ * rp_load_library_permanently
+ */
+rp_runner *rp_runner_create(uint64_t pipelines_count,
+                            const char *pipelines_path[],
+                            uint64_t pipeline_flags_count,
+                            const char *pipeline_flags[],
+                            const char *execution_directory);
+
+/**
+ * returns the number of containers registered in this pipeline
+ */
+uint64_t rp_runner_containers_count(rp_runner *runner);
+
+/**
+ * returns the container at the provided index
+ *
+ * runner cannot be null, index must be less than
+ * rp_runner_containers_count(runner)
+ */
+rp_container *rp_runner_get_container(rp_runner *runner, uint64_t index);
+
+/**
+ * returns the name of the nth container registered inside the runner
+ */
+const char *rp_container_get_name(rp_container *container);
```

Comment:
Can you group and clearly identify functions that are *class methods* of the same *class*?
Consider using [Doxygen grouping](https://www.doxygen.nl/manual/grouping.html).